docs/understanding-tracking-design/json-schema-type-casting-rules/index.md

```diff
@@ -0,0 +1,370 @@
+
+</td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": [T1, T2, ...]<br/>
+&#125;</pre>
+</td>
+<td>
+<code>VARCHAR(10)</code>
+</td>
+<td>
+Where <code>T1, T2, ..</code>. only contains <code>"boolean"</code> <b>and</b> <code>"integer"</code>
+</td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": [T1, T2, ...]<br/>
+&#125;</pre>
+</td>
+<td>
+<code>VARCHAR(4096)</code>
+</td>
+<td></td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "string",<br/>
+    &nbsp;&nbsp;"format": "date-time"<br/>
+&#125;</pre>
+</td>
+<td>
+<code>TIMESTAMP</code>
+</td>
+<td></td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "string",<br/>
+    &nbsp;&nbsp;"format": "date"<br/>
+&#125;</pre>
+</td>
+<td>
+<code>DATE</code>
+</td><td></td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "array"<br/>
+&#125;</pre>
+</td>
+<td>
+<code>VARCHAR(65535)</code>
+</td><td></td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "integer",<br/>
+    &nbsp;&nbsp;"maximum": M<br/>
+&#125;</pre>
+</td>
+<td>
+<code>SMALLINT</code>
+</td>
+<td>
+Where <code>M</code> &le; 32767
+</td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "integer",<br/>
+    &nbsp;&nbsp;"maximum": M<br/>
+&#125;</pre>
+</td>
+<td>
+<code>INT</code>
+</td>
+<td>
+Where 32767 &lt; <code>M</code> &le; 2147483647
+</td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "integer",<br/>
+    &nbsp;&nbsp;"maximum": M<br/>
+&#125;</pre>
+</td>
+<td>
+<code>BIGINT</code>
+</td>
+<td>
+Where <code>M</code> > 2147483647
+</td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "integer",<br/>
+    &nbsp;&nbsp;"enum": [E1, E2, ...]<br/>
+&#125;</pre>
+</td>
+<td>
+<code>SMALLINT</code>
+</td>
+<td>
+Where maximum <code>E*</code> &le; 32767
+</td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "integer",<br/>
+    &nbsp;&nbsp;"enum": [E1, E2, ...]<br/>
+&#125;</pre>
+</td>
+<td>
+<code>INT</code>
+</td>
+<td>
+32767 &lt; maximum <code>E*</code> &le; 2147483647
+</td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "integer",<br/>
+    &nbsp;&nbsp;"enum": [E1, E2, ...]<br/>
+&#125;</pre>
+</td>
+<td>
+<code>BIGINT</code>
+</td>
+<td>
+maximum <code>E*</code> &gt; 2147483647
+</td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "integer"<br/>
+&#125;</pre>
+</td>
+<td>
+<code>BIGINT</code>
+</td><td></td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"multipleOf": B<br/>
+&#125;</pre>
+</td>
+<td>
+<code>INT</code>
+</td><td></td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "number",<br/>
+    &nbsp;&nbsp;"multipleOf": B<br/>
+&#125;</pre>
+</td>
+<td>
+<code>DECIMAL(36,2)</code>
+</td>
+<td>
+Only works for <code>B</code>=2
```

Comment:
What happens otherwise? A different type is used? Error? Where? Failed events? What type?

Reply:
If rule does not match lookup continues down the table, as per the note above the table. Leading to DOUBLE in the next row.